### PR TITLE
Refactor git refresh system (small refactor)

### PR DESCRIPTION
**Pull request by matentzn**
This is change the Mondo build _inside_ the Mondo ingest pipeline into a method and folds mondo.owl and mondo.sssom.tsv dependency back into the normal Make dependency chain.

## Overview

This PR:
- Creates a make function to run the mondo build
- Adds the file depndencies generated by that build back into the normal Mondo dependency chain

### Documentation
Was the documentation added/updated under `docs/`?
- [ ] Yes
- [x] No, updates to the docs were not necessary after careful consideration

### QC
Was the full pipeline run before submitting this PR using `sh run.sh make build-mondo-ingest` on this branch (after 
`docker pull obolibrary/odkfull:dev`), and no errors occurred?
- [x] Yes
- [ ] No, there are no functional (code-related) changes to the pipeline in the PR, so no re-run is necessary
   
### New Packages
Were any new *Python packages* added?
- [ ] Yes, [requirements.txt.full](https://github.com/INCATools/ontology-development-kit/blob/master/requirements.txt.full) was updated and an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) submitted
- [x] No

Were any other *non-Python packages* added?
- [ ] Yes, [Dockerfile](https://github.com/INCATools/ontology-development-kit/blob/master/Dockerfile) updated and an [ODK issue](https://github.com/INCATools/ontology-development-kit/issues/new) submitted
- [x] No

### PR Review and Conversations Resolved
Has the PR been sufficiently reviewed by at least 1 team member of the Mondo Technical team and all threads resolved?
- [ ] Yes

### CODE REVIEW

**Review by twhetzel — APPROVED** [review]
LGTM

@joeflack4 can you do the various build checks you did for https://github.com/monarch-initiative/mondo-ingest/pull/551 and any others as needed for this PR to test this? This PR in general probably also needs a full feature branch build.

**Comment by matentzn** [comment]
Its a draft, this has not been tested yet :P will do it soon.

File: src/ontology/mondo-ingest.Makefile
Comment (joeflack4) [inline src/ontology/mondo-ingest.Makefile:1]: ## General notes
I see that the primary change here is the implementation of how we ensure that we check for new commits to `mondo` whenever a goal is run that depends on `mondo.sssom.tsv` or `mondo.owl`.

Instead of doing this via a `.PHONY` goal, and removing prereqs like `mondo.sssom.tsv` and inserting `$(MAKE) up-to-date-mondo.sssom.tsv`, we use `.FORCE`, and leave the prereqs as they were before. I agree that this is a cleaner approach.

Let's run the build on it once the refactor is done, but otherwise I approve of these changes.

**Comment by twhetzel** [comment]
To confirm, Nico will run the feature branch build.

**Comment by matentzn** [comment]
Something is going very wrong this this PR and I cant look into it right now; grrrrrrrrr

**Comment by twhetzel** [comment]
@matentzn last I see there is a message from you that "something is going very wrong with this PR" and that you did not have more time to look into this. What's your status now and what and who should take on next steps as needed?

**Comment by matentzn** [comment]
I will continue to work on this when the error is fixed you are working on:

```
Traceback (most recent call last):
  File "/work/src/ontology/../../src/analysis/problematic_exclusions.py", line 140, in populate_labels
    df2 = pd.read_csv(results_path).fillna('')
  File "/usr/local/lib/python3.10/dist-packages/pandas/io/parsers/readers.py", line 1026, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/usr/local/lib/python3.10/dist-packages/pandas/io/parsers/readers.py", line 620, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/usr/local/lib/python3.10/dist-packages/pandas/io/parsers/readers.py", line 1620, in __init__
    self._engine = self._make_engine(f, self.engine)
  File "/usr/local/lib/python3.10/dist-packages/pandas/io/parsers/readers.py", line 1898, in _make_engine
    return mapping[engine](f, **self.options)
  File "/usr/local/lib/python3.10/dist-packages/pandas/io/parsers/c_parser_wrapper.py", line 93, in __init__
    self._reader = parsers.TextReader(src, **kwds)
  File "parsers.pyx", line 581, in pandas._libs.parsers.TextReader.__cinit__
pandas.errors.EmptyDataError: No columns to parse from file

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/work/src/ontology/../../src/analysis/problematic_exclusions.py", line 288, in <module>
    cli()
  File "/work/src/ontology/../../src/analysis/problematic_exclusions.py", line 283, in cli
    return problematic_exclusions(**d)
  File "/work/src/ontology/../../src/analysis/problematic_exclusions.py", line 233, in problematic_exclusions
    df = populate_labels(df=df, onto_path=onto_path, prefix_map=prefix_uri_map)
  File "/work/src/ontology/../../src/analysis/problematic_exclusions.py", line 145, in populate_labels
    raise RuntimeError('No results found when trying to get labels for terms.')
RuntimeError: No results found when trying to get labels for terms.
make[1]: *** [mondo-ingest.Makefile:271: reports/icd11foundation_excluded_terms_in_mondo_xrefs.tsv] Error 1
```

Right now everything fails a bit!

**Comment by twhetzel** [comment]
@matentzn will you be doing the build for this or should someone else?

**Comment by matentzn** [comment]
I can do it, after finishing the externally managed content pipeline draft! Deadline Wednesday 21st August.

File: src/ontology/mondo-ingest.Makefile
Comment (matentzn) [inline src/ontology/mondo-ingest.Makefile:395]: @joeflack4 this might interest you. A heavily underused `make` pattern! `make` functions! Cool!

Comment (joeflack4) [reply src/ontology/mondo-ingest.Makefile:395]: You're right, it does. I only tried them once or twice and had difficulty getting the params to work. In this age of LLM I believe I'd have a much easier time.

There will be no variation in params for this function, so my guess is that it is effectively the same and there is no advantage/difference other than style, but if there is more to it, lemme know.

**Review by joeflack4 — APPROVED** [review]
I am approving now, so long as we run a build on it / make sure the recent changes run correctly.

File: src/ontology/mondo-ingest.Makefile
Comment (twhetzel) [inline src/ontology/mondo-ingest.Makefile:406]: Should `tmp/mondo.owl` be `$(TMPDIR)/mondo.owl`?

**Review by twhetzel — APPROVED** [review]
lgtm - one minor comment inline

File: src/ontology/mondo-ingest.Makefile
Comment (matentzn) [reply src/ontology/mondo-ingest.Makefile:406]: There are 29 occurrences of this in the `Makefile`.. I will do a refactor after this PR is merged.

**Pull request merged**
